lua/neo-tree/sources/filesystem/lib/fs_actions.lua

```diff
@@ -163,7 +163,7 @@ local function get_unused_name(
         local new_path = parent_path and parent_path .. utils.path_separator .. new_name or new_name
         get_unused_name(new_path, using_root_directory, name_chosen_callback)
       end
-    end)
+    end, {}, "file")
```

Comment:
`file` completion completes relative to the cwd of neo-tree, not relative to the file being renamed. For configs that use `show_path="absolute"` it's fine but the default is to prefill just the filename/dirname. I think you might be able to modify the buffer-local `'path'` option and use `file_in_path` completion to get better completion?